internal/deviations/deviations.go

```diff
@@ -72,20 +74,73 @@ import (
 	"github.com/openconfig/ondatra"
 )
 
-// lookupDutDeviations returns the deviations for the specified dut.
+func matchRegex(regex string, deviceInfo string) bool {
+	match, err := regexp.MatchString(regex, deviceInfo)
+	if err != nil {
+		log.Errorf("Error with regex match %v", err)
+	}
+	return match
+}
+
 func lookupDUTDeviations(dut *ondatra.DUTDevice) *mpb.Metadata_Deviations {
```

Comment:
Let's break this into two methods: `lookupDeviations` which returns `(*mpb.Metadata_Deviations, error)` and `mustLookupDeviations` which returns `*mpb.Metadata_Deviations` and calls `log.Exit` if `lookupDeviations` returns an error.

Reply:
Add a `lookupDeviations` which returns (*mpb.Metadata_PlatformExceptions, error) and instead of `mustLookupDeviations` updated the `lookupDUTDeviations ` to call `log.Exit` if lookupDeviations returns an error.

Let me know if this looks ok.

Reply:
Unfortunately, it makes the difference between the two functions unclear, and the Go style rule is to prefix these functions with "must": https://google.github.io/styleguide/go/decisions.html#must-functions

Was it avoided, because it requires updating lots of callers?

Reply:
thank you for pointing to this, I did not know about the `must` prefix. I updated the function names accordingly.

Yes, I did not want to change the name of the `lookupDUTDeviations` function, because it is called from every deivaiton setter and is referenced in the README file. I felt changing this name might also create confusion for PR authors who have existing PRs adding new deviations.